scripts/party.coffee

```diff
@@ -0,0 +1,55 @@
+# Description:
+#   Parties with two emoji
+#
+# Dependencies:
+#    none
+#
+# Configuration:
+#   None
+#
+# Commands:
+#    hubot party <emoji> <emoji> - ralph parties with your two emoji
+#    hubot party <emoji> <emoji> WxH - ralph parties with your two emoji in a W by H rectangle.
+#
+# Notes:
+#    None
+#
+# Author:
+#    @km42levelup
+
+DEFAULT_WIDTH = 12
+
+DEFAULT_HEIGHT = 6
+
+party = (emoji_one, emoji_two, width, height) ->
+    line_one = Array(Math.floor(width / 2) + 1).join(emoji_one + emoji_two)
+    if ((width % 2) == 1)
+        line_one += emoji_one
+
+    line_two = Array(Math.floor(width / 2) + 1).join(emoji_two + emoji_one)
+    if ((width % 2) == 1)
+        line_two += emoji_two
+
+    paragraph = ''
+    i = 0
+    while i < height
+        if ((i % 2) == 0)
+            paragraph += line_one + '\n\n'
+        else
+            paragraph += line_two + '\n\n'
+        i++
+    return paragraph
+
+
+module.exports = (robot) ->
+    robot.respond /party ([^\ ]+) ([^\ ]+)( [\d]+x[\d]+$)?/i, (msg) ->
```

Comment:
Instead of literal spaces here, could this use `\s` to match any whitespace characters? Also, may be wise to have it look for `\s+` instead of only a single space between arguments.

Reply:
And if you're using `\s`, could you conversely use `\S` for non-whitespace characters?

Reply:
k